### ISSUE TITLE
cisco.asa.asa_ogs - unable to create service-objects - invalid input error

### ISSUE THREAD

**Issue by ginigangadharan** [ansible-collections/cisco.asa#171]
##### SUMMARY

Unable to create service object using `cisco.asa.asa_ogs` module. (`object_type: service`)
(network object creation is working fine with the Same module - `object_type: network`)

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

cisco.asa.asa_ogs

##### ANSIBLE VERSION

```paste below
(ansible211) [gmadappa@gmadappa nw-automation]$ ansible --version
ansible [core 2.11.12] 
  config file = /home/gmadappa/nw-automation/ansible.cfg
  configured module search path = ['/home/gmadappa/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/gmadappa/pyvirt/ansible211/lib64/python3.10/site-packages/ansible
  ansible collection location = /home/gmadappa/nw-automation/collections
  executable location = /home/gmadappa/pyvirt/ansible211/bin/ansible
  python version = 3.10.4 (main, Mar 25 2022, 00:00:00) [GCC 11.2.1 20220127 (Red Hat 11.2.1-9)]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
(ansible211) [gmadappa@gmadappa nw-automation]$ ansible-galaxy collection list

# /home/gmadappa/nw-automation/collections/ansible_collections
Collection        Version
----------------- -------
ansible.netcommon 3.0.0  
ansible.utils     2.6.1  
cisco.asa         3.0.0  
cisco.nxos        3.0.0  
```

##### CONFIGURATION

```paste below
$ cat ansible.cfg 
[defaults]
#inventory file location
inventory = inventory 

host_key_checking = false

timeout = 5

roles_path = ./roles
COLLECTIONS_PATHS = ./collections


[persistent_connection]
#ssh_type = libssh
ssh_type = paramiko

#[ssh_connection]
#-C -o ControlMaster=auto -o ControlPersist=60s -c diffie-hellman-group1-sha1 
```

##### OS / ENVIRONMENT

Cisco ASA 9.8.1

##### STEPS TO REPRODUCE

Create a service object group in Cisco ASA device

manual way:
```
ciscoasa(config)# object-group service demo-service tcp 
ciscoasa(config-service-object-group)#

ciscoasa(config-service-object-group)# port-object eq 5000
ciscoasa(config-service-object-group)# end

ciscoasa# show running-config object-group
.
<omitted>
.
object-group service demo-service tcp
 port-object eq 5000
```

```yaml
- name: "Merge module attributes of given object-group - Service"
  cisco.asa.asa_ogs:
    config: 
    - object_type: service
      object_groups:
        - name: "{{ asa_object_group_name }}"
          services_object:
            - protocol: tcp #"{{ asa_object_group_service_protocol }}"
              destination_port:
                eq: 5000 #"{{ asa_object_group_service_port }}"
    state: merged
  when: asa_object_group_type == "service"
```

##### EXPECTED RESULTS

Successfully create the object group

```
TASK [cisco-asa-object-group-create : Merge module attributes of given object-group - Network] ***
ok: [ciscoasa]
```

##### ACTUAL RESULTS

```paste below
TASK [cisco-asa-object-group-create : Merge module attributes of given object-group - Service] ***
task path: /home/gmadappa/ansible/bau-ansible/kh-sb-nw-automation/roles/cisco-asa-object-group-create/tasks/main.yml:18
redirecting (type: action) cisco.asa.asa_ogs to cisco.asa.asa
redirecting (type: action) cisco.asa.asa_ogs to cisco.asa.asa
fatal: [ciscoasa]: FAILED! => {
    "changed": false,
    "module_stderr": "service-object tcp destination eq 5000\r\n                          ^\r\nERROR: % Invalid input detected at '^' marker.\r\n\rciscoasa(config)# ",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}

```

**Comment by justjais** [comment]
@ginigangadharan Thanks so much for raising the issue, will triage and update the issue asap

**Comment by justjais** [comment]
@ginigangadharan I've tried the play mentioned under the issue description but I am unable to reproduce the said error:
Play used:
```
- name: Bug_171 triage
  cisco.asa.asa_ogs:
    state: merged
    config:
    - object_type: service
      object_groups:
        - name: "test_object_gp"
          services_object:
            - protocol: tcp
              destination_port:
                eq: 5000
```

Output:
```
after:
  - object_groups:
    - name: test_object_gp
      services_object:
      - destination_port:
          eq: '5000'
        protocol: tcp
    object_type: service
  before: {}
  commands:
  - object-group service test_object_gp
  - service-object tcp destination eq 5000
```
Also, as I see in the manual config you're trying to configure `port-object` under service object but via play you're trying to use `destination_port` which is unexpected, for configuring `port-object` you need to use `port-object` param like:
```
- name: Service Object Groups
  cisco.asa.asa_ogs:
    state: rendered
    config:
      - object_type: service
        object_groups:
        - name: test_object_gp
          protocol: tcp
          port_object:
          - eq: 5000
```
Running this will result in following output:
```
  rendered:
  - object-group service test_object_gp tcp
  - port-object eq 5000
```
Plz [ref](https://github.com/ansible-collections/cisco.asa/blob/main/plugins/modules/asa_ogs.py#L357) the example

**Comment by ginigangadharan** [comment]
@justjais 

```
        port_object:
          - eq: 5000
```
that's is a good catch and thank you.

I was following the doc and missed that. 

```yaml
    - object_type: service
      object_groups:
        - name: O-Worker
          services_object:
            - protocol: tcp
              destination_port:
                range:
                  start: 100
                  end: 200
            - protocol: tcp-udp
              source_port:
                eq: 1234
              destination_port:
                gt: nfs
```

**Comment by ginigangadharan** [comment]
@justjais 

Noticed the below issue; the `object-group service` command is triggering twice (with and without `tcp`) which is not expected.

**Task**

```yaml
- name: Create Service Object Groups
  cisco.asa.asa_ogs:
    state: rendered
    config:
      - object_type: service
        object_groups:
        - name: test_object2
          protocol: tcp
          port_object:
          - eq: 32000
```

```shell
"rendered": [
        "object-group service test_object2",
        "object-group service test_object2 tcp",
        "port-object eq 32000"
    ]
```

- only the first command will be executed and I can see the object group `test_object2` without `tcp` !

Full output 

```shell
redirecting (type: action) cisco.asa.asa_ogs to cisco.asa.asa
ok: [ciscoasa] => {
    "changed": false,
    "invocation": {
        "module_args": {
            "config": [
                {
                    "object_groups": [
                        {
                            "description": null,
                            "group_object": null,
                            "icmp_type": null,
                            "name": "test_object2",
                            "network_object": null,
                            "port_object": [
                                {
                                    "eq": "32000",
                                    "range": null
                                }
                            ],
                            "protocol": "tcp",
                            "protocol_object": null,
                            "security_group": null,
                            "service_object": null,
                            "services_object": null,
                            "user_object": null
                        }
                    ],
                    "object_type": "service"
                }
            ],
            "running_config": null,
            "state": "rendered"
        }
    },
    "rendered": [
        "object-group service test_object2",
        "object-group service test_object2 tcp",
        "port-object eq 32000"
    ]
}
```

And indeed it will fail as below when I put `state: merged` !

```shell
TASK [cisco-asa-object-group-create : Create Service Object Groups] ********************************************
task path: /home/gmadappa/ansible/bau-ansible/kh-sb-nw-automation/roles/cisco-asa-object-group-create/tasks/main.yml:38
redirecting (type: action) cisco.asa.asa_ogs to cisco.asa.asa
redirecting (type: action) cisco.asa.asa_ogs to cisco.asa.asa
fatal: [ciscoasa]: FAILED! => {
    "changed": false,
    "module_stderr": "port-object eq 32000\r\n                       ^\r\nERROR: % Invalid input detected at '^' marker.\r\n\rciscoasa(config)# ",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"
}

```

Am I missing anything ?

**Comment by justjais** [comment]
@ginigangadharan yes, that's a bug and was raised [here](https://github.com/ansible-collections/cisco.asa/issues/166), also [PR](https://github.com/ansible-collections/cisco.asa/pull/173) is raised to fix the said issue and it'd be great if you can verify the PR fix and confirm if it resolves the issue.

**Comment by ginigangadharan** [comment]
@justjais thank you.

**Comment by justjais** [comment]
Sure, and the PR with the fix is merged and is available with the `3.1.0` release.
